controllers/managedocs_controller.go

```diff
@@ -920,6 +936,41 @@ func (r *ManagedOCSReconciler) reconcileOnboardingValidationSecret() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcileCSIKMSConnectionDetailsConfigMap() error {
+	if r.DeploymentType != consumerDeploymentType {
+		r.Log.Info(fmt.Sprintf("Non consumer deployment, skipping reconcile for %s config map", csiKMSConnectionDetailsConfigMapName))
+		return nil
+	}
+	r.Log.Info(fmt.Sprintf("Reconciling %s configMap", csiKMSConnectionDetailsConfigMapName))
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.csiKMSConnectionDetailsConfigMap, func() error {
+		if err := r.own(r.csiKMSConnectionDetailsConfigMap); err != nil {
+			return err
+		}
+
+		encryptionMetadata := struct {
+			EncryptionKMSType string `json:"encryptionKMSType"`
+			SecretName        string `json:"secretName"`
+		}{
+			EncryptionKMSType: "aws-sts-metadata",
+			SecretName:        "ceph-csi-aws-credentials", // user needs to create a secret with this name
+		}
+
+		data, _ := json.Marshal(encryptionMetadata)
+
+		r.csiKMSConnectionDetailsConfigMap.Data = map[string]string{
+			"aws-sts-metadata-test": string(data), // encryption-with-aws-kms will be used by customer to create storageClassClaim
+		}
```

Comment:
- Why do we need an anonymous struct definition for two string fields? why can't we just use a map?
- If you are ignoring the error for marshal (which makes sense here), why not use `utils.toJsonOrDie()` and inline it? 
```suggestion
		r.csiKMSConnectionDetailsConfigMap.Data = map[string]string{
			"aws-sts-metadata-test": utils.ToJsonOrDie(map[string]string {
				"encryptionKMSType": "aws-sts-metadata",
				"secretName": "ceph-csi-aws-credentials", // user needs to create a secret with this name
			}),
		}
```

Reply:
updated